src/validation.rs

```diff
@@ -0,0 +1,40 @@
+use axum::{async_trait, extract::{FromRequest, rejection::JsonRejection}, Json, RequestExt, response::IntoResponse};
+use http::{Request, StatusCode};
+use validator::{Validate, ValidationErrors};
+
+pub enum ValidatedJsonRejection {
+    JsonRejection(JsonRejection),
+    ValidationRejection(ValidationErrors)
+}
+
+impl IntoResponse for ValidatedJsonRejection {
+    fn into_response(self) -> axum::response::Response {
+        match self {
+            Self::JsonRejection(rejection) => rejection.into_response(),
```

Comment:
Notice that this may return status codes not included in swagger (e.g. 415)

Reply:
Nice catch